File: frontend/src/types/workspace.ts
import type { Task } from "./task";

export enum WorkspaceType {
  PROJECT_MANAGEMENT = "PROJECT_MANAGEMENT",
  FIELD_OPERATIONS = "FIELD_OPERATIONS",
}

export interface Workspace {
  id: number;
  name: string;
  description?: string;
  icon?: string;
  color?: string;
  owner_id: number;
  workspace_type: WorkspaceType;
  created_at: string;
  updated_at: string;
}

export interface WorkspaceMember {
  id: number;
  user_id: number;
  username: string;
  email: string;
  role: "owner" | "manager" | "member";
  full_name?: string;
  avatar_url?: string;
  joined_at: string;
}

export interface WorkspaceDetail extends Workspace {
  members: WorkspaceMember[];
}

export interface Project {
  id: number;
  name: string;
  description?: string;
  workspace_id: number;
  color?: string;
  icon?: string;
  created_at: string;
  updated_at: string;
}

export interface ProjectDetail extends Project {
  tasks: Task[];
}

File: backend/app/schemas/workspace.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from app.models.workspace import WorkspaceType


class WorkspaceBase(BaseModel):
    name: str
    description: Optional[str] = None
    icon: Optional[str] = None
    color: Optional[str] = None


class WorkspaceCreate(WorkspaceBase):
    workspace_type: WorkspaceType = WorkspaceType.PROJECT_MANAGEMENT


class WorkspaceUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    icon: Optional[str] = None
    color: Optional[str] = None
    workspace_type: Optional[WorkspaceType] = None


class WorkspaceMemberResponse(BaseModel):
    id: int
    user_id: int
    username: str
    email: str
    full_name: Optional[str]
    avatar_url: Optional[str]
    system_role: str
    joined_at: datetime

    class Config:
        from_attributes = True


class WorkspaceProjectResponse(BaseModel):
    id: int
    name: str
    description: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class WorkspaceResponse(WorkspaceBase):
    id: int
    owner_id: int
    workspace_type: WorkspaceType
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class WorkspaceDetailResponse(WorkspaceResponse):
    members: List[WorkspaceMemberResponse] = []
    # projects: List[WorkspaceProjectResponse] = []

    class Config:
        from_attributes = True
